tests/4.5/task/test_task_Mergeable.c

```diff
@@ -0,0 +1,67 @@
+//===-- test_task_Mergeable.c ------------------------------------------------===//
+//
+// OpenMP API Version 4.5 Nov 2015
+//
+// Description
+// Test case description: This test case demonstrates that mergeable clause
+// help thread avoid task creation can queing. The if condition turns out
+// false for thread 10, this enables the thread 10 acquire more tasks from
+// the queue and run resulting in higher count value.
+// 
+//===----------------------------------------------------------------------===//
+
+
+#include <stdio.h>
+#include <stdbool.h>
+#include <unistd.h>
+#include "omp.h"
+#include "ompvv.h"
+#define THREADS 64
+
+
+bool IfTstPassed = true;
+int arrA[THREADS];
+int GlobalThrdId = -1;
+int count = 0;
+
+/*void WaitFunc() {
+  usleep(1000);
+}*/
+
+void WaitFunc2() {
+  usleep(1*1000*1000);
+}
+
+void foo() {
+  int ThrdId = omp_get_thread_num();
+
+#pragma omp task mergeable if(ThrdId != 10)
+  {
```

Comment:
I am not sure what this task does. All threads except thread 10 call this task -- essentially there are (in case there are 64 threads) 63 mergeable tasks. All these threads try to update a shared variable "GlobalThrdId" without atomic.

Reply:
Added new code to test mergeable with if clause...